Code/Framework/AzFramework/AzFramework/Archive/ArchiveFindData.h

```diff
@@ -44,18 +44,26 @@ namespace AZ::IO
         ArchiveFileIterator operator++();
         ArchiveFileIterator operator++(int);
 
+        bool operator==(const AZ::IO::ArchiveFileIterator& rhs) const;
+
         explicit operator bool() const;
 
         inline static constexpr size_t FilenameMaxLength = 256;
         AZStd::fixed_string<FilenameMaxLength> m_filename;
         FileDesc m_fileDesc;
         AZStd::intrusive_ptr<FindData> m_findData{};
+        size_t m_hash; // hash value used for storage in unordered_set
```

Comment:
This should be initialized to zero, as the defaulted default constructor above will not initialize this to a value.

The constructor that you updated will work, but the not the default

Reply:
Ah right, good catch.

Reply:
There is also the lot bigger issue that the the hash needs to be kept in sync with the `m_filename`, which it can't be automatically, as the `m_filename` is the public member of the struct.

It would be safer to not store of the hash value and have it be calculated on demand.

Alternatively if for performance reasons the hash needs to be stored off, then the `m_filename` member should made private and placed behind a setter, that would calculate the hash whenever it is set.
Following the alternative recommendation though would require updating all places that queries  the `m_filename` member, with a call to a getter.

Reply:
Completely valid point and had certainly crossed my mind.  I had initially tried to delete the default c-tor and existing usage paints a pretty clear picture setting the filename after construction is not happening but neither really solve the problem.

I'll run some more tests, even throw in your change to the PathView specialization just in case, to confirm if that wider change is even worth it.

Reply:
Turns out the perf difference here isn't enough to warrant the additional work to ensure the cached hash is in sync.  The average timings I was getting for the tests were within a few milliseconds of each other.  Will revert back to on demand hashing.